server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueue.java

```diff
@@ -180,6 +180,14 @@ public MailQueueItem deQueue() throws MailQueueException {
         GetResponse getResponse = pollChannel();
         MailDTO mailDTO = toDTO(getResponse);
         Mail mail = toMail(mailDTO);
+        if (mailQueueView.isDeleted(mail).join()) {
+            try {
+                rabbitClient.ack(getResponse.getEnvelope().getDeliveryTag());
+            } catch (IOException e) {
+                throw new MailQueueException("Failed to ack", e);
+            }
+            return deQueue();
```

Comment:
it makes the stack growing needlessly (I don't think it has tail-call optimization), you should split this method in two, a `Optional<GetResponse> tryDequeue` method and the dequeue method would loop until tryDequeue return a not-deleted mail.

Reply:
Didn't apply after rebase.